plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt

```diff
@@ -106,30 +107,27 @@ class WebviewBrowser(val project: Project) : LoginBrowser(project, WebviewBrowse
                 this.prepareBrowser(BrowserState(FeatureId.Q, false))
             }
 
-            "loginBuilderId" -> {
-                val scope = CODEWHISPERER_SCOPES + Q_SCOPES
-                runInEdt {
-                    Login.BuilderId(scope, onPendingAwsId).loginBuilderId(project)
-                    // TODO: telemetry
+            "selectConnection" -> {
+                val connId = jsonTree.get("connectionId").asText()
+                this.selectionSettings[connId]?.let { settings ->
+                    settings.onChange(settings.currentSelection)
                 }
             }
 
+            "loginBuilderId" -> {
+                loginBuilderId(CODEWHISPERER_SCOPES + Q_SCOPES)
```

Comment:
Should we combine these scopes as Q_SCOPES now?

Reply:
yes, we should, let's do it in another PR